### PR TITLE
Improve docker build

### PR DESCRIPTION
**Pull request by sagikazarmark**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets |
| License         | Apache 2.0

### What's in this PR?
Improve docker build
Push images to ghcr.io

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,9 +1,13 @@
 name: CI
 
 on:
+  schedule:
```
Comment (bonifaido) [inline .github/workflows/ci.yml:4]: Why do we need this schedule?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -196,24 +200,70 @@ jobs:
...
+          elif [[ $GITHUB_REF == refs/heads/* ]]; then
+            VERSION=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+            if [ "${{ github.event.repository.default_branch }}" = "$VERSION" ]; then
+              VERSION=edge
+            fi
+          elif [[ $GITHUB_REF == refs/pull/* ]]; then
+            VERSION=pr-${{ github.event.number }}
+          fi
+          TAGS="${DOCKER_IMAGE}:${VERSION}"
+          if [[ $VERSION =~ ^v?[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
```
Comment (pregnor) [inline .github/workflows/ci.yml:232]: **Question**: shouldn't we drop the `v` prefix as we are not using `v1.2.3` tags in this repository, but `1.2.3` instead?
Or are we planning to change the tag naming strategy in accordance to this change?

According to the tag matching expression [earlier](https://github.com/banzaicloud/pipeline/pull/3207/files#diff-e9f950f17198d3d5e3122a44230a09b9R10) I'm supposing this is just a copy-paste mistake.

Comment (pregnor) [inline .github/workflows/ci.yml:232]: **Question**: why are we limiting the number of version digits to 3?
Realistically it should be enough (so this is not a hard change request), but there is always a possiblity to use dates as valid semver version which is not matched by this expression.

Also at the GHA [event trigger](https://github.com/banzaicloud/pipeline/pull/3207/files#diff-e9f950f17198d3d5e3122a44230a09b9R10) we defined it unbound to the number of digits.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sagikazarmark) [reply .github/workflows/ci.yml:4]: For nightly builds to detect flaky tests or issues caused by automaticly updated components.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sagikazarmark) [reply .github/workflows/ci.yml:232]: Note the questionmark after 'v'.  It makes the step more portable.

Comment (sagikazarmark) [reply .github/workflows/ci.yml:232]: That's a copy-paste issue

Comment (pregnor) [reply .github/workflows/ci.yml:232]: Yeah, I totally missed it when read, apologies.

Comment (sagikazarmark) [reply .github/workflows/ci.yml:232]: No worries 🙂

Comment (sagikazarmark) [reply .github/workflows/ci.yml]: Fixed

**Review by pregnor — APPROVED** [review]

**Review by bonifaido — APPROVED** [review]

**Pull request merged**